### ISSUE TITLE
Refreshing analyzer list when plugin is enabled/disabled

### ISSUE THREAD

**Issue by astrelsky** [NationalSecurityAgency/ghidra#1930]
I have a few analyzers which depend on a plugin being enabled. It doesn't seem like the AutoAnalysisManager refreshes it's list when this occurs. Is there a convenient ay to have an Analyzer only appear in the list when a specific plugin is enabled?

**Comment by dragonmacher** [comment]
I cannot recall having needed this functionality before, but it is an interesting idea.   This may just be a semantic issue, but your analyzer should not depend on plugins, but rather on services.

It appears that you can accomplish what you need by controlling enablement via `Options`:

```
Options analysisOptions = program.getOptions(Program.ANALYSIS_PROPERTIES);
analysisOptions.setBoolean("Analyzer Name", true/false);
```

See `ghidra.app.plugin.core.analysis.AnalysisPanel` for how this is used to setup the default enablement of analyzers in the UI.

**Comment by astrelsky** [comment]
> I cannot recall having needed this functionality before, but it is an interesting idea.   This may just be a semantic issue, but your analyzer should not depend on plugins, but rather on services.
> 
> It appears that you can accomplish what you need by controlling enablement via `Options`:
> 
> ```
> Options analysisOptions = program.getOptions(Program.ANALYSIS_PROPERTIES);
> analysisOptions.setBoolean("Analyzer Name", true/false);
> ```
> 
> See `ghidra.app.plugin.core.analysis.AnalysisPanel` for how this is used to setup the default enablement of analyzers in the UI.

You are correct about the dependency on a service as opposed to a plugin. It should only be available if the service is being provided. It is simple enough to check whether or not it is being provided however `Analyzer.canAnalyze` is only invoked when the AutoAnalysisManager is constructed.

**Comment by dragonmacher** [comment]
What I suggested would not quite get what you want, as even if disabled, the user could still attempt to select the analyzer, which will not work if the correct plugins are not loaded.   It seems the best you can do with the current setup is to boldly announce your dependency in the analzer's description panel, indicating that they will not actually run if the specific plugins are not installed.
